### ISSUE TITLE
Tipado: Documentar con JSDoc // Pasar a typescript

### ISSUE THREAD

**Issue by lazcanoluca** [FdelMazo/FIUBA-Map#187]
Así tenemos sweet sweet intelisense en todo el proyecto, teniendo bien definidos los tipos de datos. Me parece que suma muchísimo, sobre todo para familiarizarse con el código y tener bien definidos los tipos (usuario, carrera, materia, etc).

Si aceptan la propuesta, yo ya la estoy trabajando, así que me pueden asignar.

**Comment by FdelMazo** [comment]
Tener el proyecto documentado y tipado me parece genial. 

Entre jsdoc y typescript me parece que prefiero typescript (por flexibilidad y ser mas moderno y mas usado), pero si ya tenes mucho hecho con jsdoc, barbaro, vamos con eso!

**Comment by FdelMazo** [comment]
Gracias por la idea!

**Comment by lazcanoluca** [comment]
Sí, claro que si estás abierto a ir metiéndole TS, me parece infinitamente mejor. Sugerí JSDoc como para empezar. Ya está TS como dep? O habría que instalarlo?

**Comment by FdelMazo** [comment]
Typescript habria que agregarlo desde cero. Pero puede ser un buen esfuerzo! Si no lo arrancas vos, el finde largo le meto al menos un puntapie

**Comment by lazcanoluca** [comment]
Buenas! Sólo para comentar que empecé con esto, no sé qué tan rápido avance porque se me vienen los parciales pero bueno.

¿Abro una PR en draft para ir trackeando el progreso? No sé bien cómo se maneja eso en un proyecto colaborativo 👀.

¿Me habías mandado invite para colaborar? Se me re pasó y no la acepté a tiempo 🤦🏻‍♂️.

PD: ¿hay algún motivo particular por el cual las correlativas están almacenadas como string, y no array?

**Comment by salluzziluca** [comment]
Buenass, todo bien? Me gustaria colabrar, por lo que entendí la idea es ir pasando toda la codebase a ts, no? Justo andaba aprendiendo y puede ser una buena forma de familiarizarme con el lenguaje. El unico que tema es que nos vamos a confundir entre Luca y Luca ajajajja

**Comment by lazcanoluca** [comment]
Perfecto Fede! Gracias!

> Las correlativas son un string porque cuando arranque el proyecto en vez de usar un json usaba un csv como datos, y no se me ocurrio nada mejor. Basicamente, legacy code. Un array me parece mucho mas apropiado!

Jajaja está perfecto, tiene sentido.

@salluzziluca grande! Voy a abrir una draft PR, con algunos tipos que definí. De ahí los mejoramos para que modelen lo mejor posible todo. Con esa base, podemos ir adaptando el resto sin miedo, en una branch aparte. Sounds good @FdelMazo ?
